### ISSUE TITLE
safely exit from model run when solver errors N number of times

### ISSUE THREAD

**Issue by vjd** [PumasAI/Pumas.jl#758]
There are situations when the optimizer provides values to the solver that results in messages like
`Large Maxiters needed` or similar ones.. 
We need a way to tell the solver to stop searching and safely exit the process while telling the optimizer that this is an unsuccessful run.

**Comment by andreasnoack** [comment]
While it *might* be an unsuccessful run, the warning might also happen as part of a line search such that the next step could be fine. Hence, it's fine to return `Inf` when we hit this case. I think the main problem is that we keep evaluating new subjects after finishing a subject that produces the warning. That is not necessary since we already know that the objective will be `Inf`. Hence, this issue is actually the same as https://github.com/PumasAI/Pumas.jl/issues/504 which we closed a little too early. Indeed, the closing PR fixed the issue when just calling `deviance/marginall_nll` on a `Population` but we don't call this function during estimation so we need a similar change in the `fit` method.

**Comment by vjd** [comment]
The suggestion was to see if there are 20 or more such successive warnings, it is best to exit early. Thanks for cross-reference.

**Comment by andreasnoack** [comment]
Now that we are using AD for all gradient components this might no longer be a problem. Personally, I haven't seen since the switch.

**Comment by vjd** [comment]
I agree. I haven't seen too. Perhaps an important note to make in the documentation
